src/cloudinaryAdapter/index.ts

```
@@ -0,0 +1,194 @@
+    if (cloudinaryOptions.enabled === false) {
+      return incomingConfig
+    }
+
+    const adapter = cloudinaryStorageInternal(cloudinaryOptions)
+
+    // Add adapter to each collection option object
+    const collectionsWithAdapter: CloudStoragePluginOptions['collections'] = Object.entries(
+      cloudinaryOptions.collections,
+    ).reduce(
+      (acc, [slug, collOptions]) => ({
+        ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)